tests/unit/test_add_contract_to_yml.py

```diff
@@ -54,12 +56,41 @@ def model() -> ModelNode:
     )
 
 
+@pytest.fixture
+def all_cap_model() -> ModelNode:
+    return ModelNode(
+        database=None,
+        resource_type=NodeType.Model,
+        checksum=FileHash("foo", "foo"),
+        schema="foo",
+        name=model_name,
+        package_name="foo",
+        path="models/_models.yml",
+        original_file_path=f"models/{model_name}.sql",
+        unique_id="model.foo.foo",
+        columns={
+            "ID": ColumnInfo.from_dict({"name": "ID", "description": "this is the id column"}),
+            "COLLEAGUE": ColumnInfo.from_dict(
+                {"name": "COLLEAGUE", "description": "this is the colleague column"}
+            ),
+        },
+        fqn=["foo", "foo"],
+        alias="foo",
+    )
+
+
 @pytest.fixture
 def change(project, model) -> ResourceChange:
     contractor = Contractor(project=project)
     return contractor.generate_contract(model)
 
 
+@pytest.fixture
+def all_cap_model_change(project, all_cap_model) -> ResourceChange:
+    contractor = Contractor(project=project)
+    return contractor.generate_contract(all_cap_model)
```

Comment:
This isn't terrible, but I see what you mean by code smell. I suspect that this fixture could follow a factory pattern instead, in which the fixture is a function that itself generates or returns models that have different capitalization configurations.